src/server/pfs/server/api_server.go

```diff
@@ -318,8 +320,50 @@ func (a *apiServer) PutFile(putFileServer pfs.API_PutFileServer) (retErr error)
 	return nil
 }
 
+type putFileHelperLog struct {
+	Service  string `json:"service,omitempty"`
+	Method   string `json:"method,omitempty"`
+	Request  string `json:"request,omitempty"`
+	Duration string `json:"duration,omitempty"`
+	Error    error  `json:"error,omitempty"`
+	FilePath string `json:"filepath,omitempty"`
+	ObjPath  string `json:"objpath,omitempty"`
+}
+
+func putFileLogHelper(request *pfs.PutFileRequest, err error, duration time.Duration, filePath string, objPath string) {
+	requestString := ""
+	if request != nil {
+		requestString = request.String()
+	}
+	l := &putFileHelperLog{
+		Service:  "pfs.API",
+		Method:   "putFileObjHelper",
+		Request:  requestString,
+		Duration: duration.String(),
+		Error:    err,
+		FilePath: filePath,
+		ObjPath:  objPath,
+	}
+	logJSON, err := json.Marshal(l)
```

Comment:
Just supplying the struct results in the following log line:

```
2017-02-27T18:06:47Z INFO  &{pfs.API putFileObjHelper file:<commit:<repo:<name:"debug" > id:"j" > path:"test" > file_type:FILE_TYPE_REGULAR url:"s3://tests3recursiveputs/test" recursive:true  529.741697ms <nil> test/29388.txt test/29388.txt}
```

I think the `protolion` package automatically does the marshaling but then I'd have to back the struct w a proto object. I almost [could've used their rpc error method](https://github.com/pachyderm/pachyderm/blob/master/src/server/vendor/go.pedge.io/proto/rpclog/protorpclog.go#L79) except for this new log type I needed a few new fields -- the `filePath` and objPath` info.